src/Control/Email/Email.php

```diff
@@ -276,14 +276,26 @@ public function setSwiftMessage($swiftMessage)
         $dateTime = new DateTime();
         $dateTime->setTimestamp(DBDatetime::now()->getTimestamp());
         $swiftMessage->setDate($dateTime);
-        if (!$swiftMessage->getFrom() && ($defaultFrom = $this->config()->get('admin_email'))) {
-            $swiftMessage->setFrom($defaultFrom);
+        if (!$swiftMessage->getFrom()) {
+            $swiftMessage->setFrom($this->getDefaultFrom());
         }
         $this->swiftMessage = $swiftMessage;
 
         return $this;
     }
 
+    /**
+     * @return string
+     */
+    private function getDefaultFrom(): string
+    {
+        $defaultFrom = $this->config()->get('admin_email');
+        if ($defaultFrom) {
+            return $defaultFrom;
+        }
+        return sprintf('no-reply@%s', parse_url(Director::host(), PHP_URL_HOST));
```

Comment:
Something like this..

```suggestion
        $defaultFrom = $this->config()->get('admin_email');
        if (!$defaultFrom) {
            $defaultFrom = sprintf('no-reply@%s', parse_url(Director::host(), PHP_URL_HOST));
        }
        
        $this->extend('updateDefaultFrom', $defaultFrom);
        
        return $defaultFrom ;
```

Reply:
Updated